File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/model/JcrFullManager.java
package solutions.digamma.damas.jcr.model;

import solutions.digamma.damas.entity.CrudManager;
import solutions.digamma.damas.entity.Entity;
import solutions.digamma.damas.content.PathFinder;
import solutions.digamma.damas.entity.SearchEngine;

/**
 * JCR full manager.
 *
 * @author Ahmad Shahwan
 */
abstract public class JcrFullManager<T extends Entity>
    extends JcrCrudManager<T>
    implements CrudManager<T>, SearchEngine<T>, PathFinder<T> {
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/model/JcrFullManager.kt
package solutions.digamma.damas.jcr.model

import solutions.digamma.damas.entity.CrudManager
import solutions.digamma.damas.entity.Entity
import solutions.digamma.damas.content.PathFinder
import solutions.digamma.damas.search.Filter
import solutions.digamma.damas.search.SearchEngine

/**
 * JCR full manager.
 *
 * @author Ahmad Shahwan
 */
internal abstract class JcrFullManager<T : Entity> :
        JcrCrudManager<T>(),
        CrudManager<T>,
        SearchEngine<T, Filter>,
        PathFinder<T>
